src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestRetryFailedManagedIndexAction.kt

```diff
@@ -0,0 +1,174 @@
+
+        override fun processResponse(clusterStateResponse: ClusterStateResponse) {
+            val state = clusterStateResponse.state
+            populateList(state, startState)
+
+            val builder = channel.newBuilder().startObject()
+            if (listOfIndexMetaData.isNotEmpty()) {
+                val updateManagedIndexMetaDataRequest = UpdateManagedIndexMetaDataRequest(listOfIndexMetaData)
+
+                try {
+                    client.execute(UpdateManagedIndexMetaDataAction, updateManagedIndexMetaDataRequest,
+                        object : RestResponseListener<AcknowledgedResponse>(channel) {
+                            override fun buildResponse(acknowledgedResponse: AcknowledgedResponse): RestResponse {
+                                if (acknowledgedResponse.isAcknowledged) {
+                                    builder.field(UPDATED_INDICES, listOfIndexMetaData.size)
+                                } else {
+                                    failedIndices.addAll(listOfIndexMetaData.map {
+                                        FailedIndex(it.first.name, it.first.uuid, "failed to update IndexMetaData")
+                                    })
+                                }
+                                buildInvalidIndexResponse(builder)
+                                return BytesRestResponse(RestStatus.OK, builder.endObject())
+                            }
+                        }
+                    )
+                } catch (e: ClusterBlockException) {
+                    failedIndices.addAll(listOfIndexMetaData.map {
+                        FailedIndex(it.first.name, it.first.uuid, "failed to update with ClusterBlockException. ${e.message}")
+                    })
+                    buildInvalidIndexResponse(builder)
+                    channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+                }
+            } else {
+                buildInvalidIndexResponse(builder)
+                channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+            }
+        }
+
+        private fun buildInvalidIndexResponse(builder: XContentBuilder) {
+            if (failedIndices.isNotEmpty()) {
+                builder.field(FAILURES, true)
+                builder.startArray(FAILED_INDICES)
+                for (failedIndex in failedIndices) {
+                    builder.startObject()
+                    builder.field("index_name", failedIndex.name)
+                    builder.field("index_uuid", failedIndex.uuid)
+                    builder.field("reason", failedIndex.reason)
+                    builder.endObject()
+                }
+                builder.endArray()
+            } else {
+                builder.field(FAILURES, false)
+            }
+        }
+
+        private fun populateList(state: ClusterState, startState: String?) {
```

Comment:
np: This is essentially a big else if chain, could probably clean it up to something like:
```
private fun populateList(state: ClusterState, startState: String?) {
    for (indexMetaDataEntry in state.metaData.indices) {
        val indexMetaData = indexMetaDataEntry.value
        val managedIndexMetaData = indexMetaData.getManagedIndexMetaData()
        when {
            indexMetaData.getPolicyName() == null -> failedIndices.add(...)
            managedIndexMetaData == null -> failedIndices.add(...)
            managedIndexMetaData.failed != true -> failedIndices.add(...)
            else -> listOfIndexMetaData.add(...)
        }
    }
}
```